x/marker/types/authz_test.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+)
+
+var (
+	coin1000 = sdk.NewCoin("stake", sdk.NewInt(1000))
+	coin500  = sdk.NewCoin("stake", sdk.NewInt(500))
+)
+
+func TestMarkerTransferAuthorization(t *testing.T) {
```

Comment:
Nit: You don't need to change anything in here. This comment is mostly for future stuff.

I prefer to have tests split out by function, and then use `t.Run(...)` to identify specific test cases on that function.
1. It makes it easier to find where a function is being unit tested.
2. When something fails, it's easier to identify what the failure is.
3. If multiple parts break, it tells you all of them rather than just the first failure. That makes fixing things easier since A) It's easier to identify the root problem. B) Even when it only takes 10 seconds to run the unit tests, it's still an interruption of work flow. C) When you don't know everything that's broken, it's impossible to know when you might be done fixing things.

Reply:
Updated to use functions.